JOIEnergy/Controllers/CostCalculatorController.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Linq;
+using JOIEnergy.Enums;
+using JOIEnergy.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace JOIEnergy.Controllers
+{
+    [Route("cost-calculators")]
+    public class CostCalculatorController : Controller
+    {
+        private readonly ICostCalculatorService _costCalculatorService;
+        private readonly IAccountService _accountService;
+        private readonly IPricePlanService _pricePlanService;
+        private readonly IMeterReadingService _meterReadingService;
+
+        public CostCalculatorController(ICostCalculatorService costCalculatorService,
+            IAccountService accountService,
+            IPricePlanService pricePlanService,
+            IMeterReadingService meterReadingService)
+        {
+            _pricePlanService = pricePlanService;
+            _meterReadingService = meterReadingService;
+            _accountService = accountService;
+            _costCalculatorService = costCalculatorService;
+        }
+
+        [HttpGet("cost-of-last-week/{smartMeterId}")]
+        public IActionResult GetCostLastWeekByMeterId(string smartMeterId)
+        {
+            var supplier = _accountService.GetPricePlanIdForSmartMeterId(smartMeterId);
+            if (supplier == Supplier.NullSupplier)
+            {
+                return NotFound(string.Format("Smart Meter ID ({0}) not found", smartMeterId));
+            }
+            var pricePlan = _pricePlanService.GetPricePlanBySupplier(supplier);
+            if (pricePlan == null)
+            {
+                return NotFound(string.Format("Price Plan for Supplier ({0}) not found", supplier.ToString()));
+            }
+            var today = DateTime.Now;
+            var thisSunday = today.AddDays(-(int)today.DayOfWeek).Date;
+            var lastSunday = thisSunday.AddDays(-7).Date;
+            var lastWeekReadings = _meterReadingService.GetReadingsInPeriod(smartMeterId, lastSunday, thisSunday);
+            if (!lastWeekReadings.Any())
+            {
+                return Ok(0);
+            }
+            return Ok(_costCalculatorService.CalculateCost(pricePlan, lastWeekReadings));
```

Comment:
Should move this logic to the service layer. Keep your controller action concise.